index.js

```diff
@@ -12,7 +12,26 @@ function isNegative(pattern) {
 	return pattern[0] === '!';
 }
 
+function isString(value) {
+	return typeof value === 'string';
+}
+
+function assertPatternsInput(patterns) {
+	if (isString(patterns)) {
+		// bail early, the rest will easier this way
+		return;
+	}
+
+	if (!Array.isArray(patterns) || !patterns.every(isString)) {
+		throw new TypeError('patterns must be a string or an array of strings');
+	}
+}
+
 function generateGlobTasks(patterns, opts) {
+	// make sure input is correct, so that we can safely parse
+	// it in this task
+	assertPatternsInput(patterns);
+
 	var globTasks = [];
 
 	patterns = arrify(patterns);
```

Comment:
@catdad If `null` and `undefined` are considered as errors, then I don't see any benefit of using [`arrify`](https://github.com/sindresorhus/arrify/blob/v1.0.1/index.js) at all:

```js
function generateGlobTasks(patterns, opts) {
  	var globTasks = [];
  
  	patterns = Array.isArray(patterns) ? patterns : [patterns];
	assertPatternsInput(patterns);
```

(and `assertPatternsInput` could be simplified, as already suggested)

Furthermore, as we use (self-explanatory) speaking names, personally, I see no advantage of comments and would remove them.